packages/dev/serializers/src/glTF/2.0/glTFExporter.ts

```diff
@@ -1267,1083 +680,879 @@ export class _Exporter {
-                };
-            } else if (glTFCamera.type === CameraType.ORTHOGRAPHIC) {
-                const halfWidth = camera.orthoLeft && camera.orthoRight ? 0.5 * (camera.orthoRight - camera.orthoLeft) : camera.getEngine().getRenderWidth() * 0.5;
-                const halfHeight = camera.orthoBottom && camera.orthoTop ? 0.5 * (camera.orthoTop - camera.orthoBottom) : camera.getEngine().getRenderHeight() * 0.5;
-                glTFCamera.orthographic = {
-                    xmag: halfWidth,
-                    ymag: halfHeight,
-                    znear: camera.minZ,
-                    zfar: camera.maxZ,
-                };
+            const byteStride = vertexBuffers[0].byteStride;
+            if (vertexBuffers.some((vertexBuffer) => vertexBuffer.byteStride !== byteStride)) {
+                throw new Error("Vertex buffers pointing to the same buffer must have the same byte stride");
             }
 
-            cameraMap.set(camera, this._cameras.length);
-            this._cameras.push(glTFCamera);
-        });
+            const bytes = DataArrayToUint8Array(data).slice();
+
+            // Apply conversions to buffer data in-place.
+            for (const vertexBuffer of vertexBuffers) {
+                const { byteOffset, byteStride, type, normalized } = vertexBuffer;
+                const size = vertexBuffer.getSize();
+                const meshes = vertexBufferToMeshesMap.get(vertexBuffer)!;
+                const maxTotalVertices = meshes.reduce((max, current) => {
+                    return current.getTotalVertices() > max ? current.getTotalVertices() : max;
+                }, -Number.MAX_VALUE); // To ensure nothing is missed when enumerating, but may not be necessary.
+
+                switch (vertexBuffer.getKind()) {
+                    // Normalize normals and tangents.
+                    case VertexBuffer.NormalKind:
+                    case VertexBuffer.TangentKind: {
+                        EnumerateFloatValues(bytes, byteOffset, byteStride, size, type, maxTotalVertices * size, normalized, (values) => {
+                            const invLength = 1 / Math.sqrt(values[0] * values[0] + values[1] * values[1] + values[2] * values[2]);
+                            values[0] *= invLength;
+                            values[1] *= invLength;
+                            values[2] *= invLength;
+                        });
+                        break;
+                    }
+                    // Convert StandardMaterial vertex colors from gamma to linear space.
+                    case VertexBuffer.ColorKind: {
+                        const stdMaterialCount = meshes.filter((mesh) => mesh.material instanceof StandardMaterial || mesh.material == null).length;
+
+                        if (stdMaterialCount == 0) {
+                            break; // Buffer not used by StandardMaterials, so no conversion needed.
+                        }
+
+                        // TODO: Implement this case.

```

Comment:
@alexchuber , I think the main decision on this one is that it will be implemented in the future depending on how often this shows up in the community right?

Reply:
Yes, we decided not to implement this for now because it would be tedious.
I suppose we could remove the TODO, since I think this is low-priority & the logged warning looks sufficient.